### DISCUSSION TITLE
FR: Facilitate for more than one vacuum

### DISCUSSION THREAD

**Discussion by Skeletorjus** [sca075/mqtt_vacuum_camera#10] [Ideas]
First of all thank you for this project!

I'm currently using this with a S5, but am looking to use it with a gen1 as well (as soon as I update it from an ancient version of Valetudo). To make this easier it would be nice if the integration took the vacuum name into account when creating the entity name. Maybe the entity also could be registered within the vacuum device..?
Also, the snapshot filename is hardcoded to be named valetudo_snapshot.png. Could this be configurable so the different vacuums get unique filenames? A closely related idea is for the config flow to give a heads up of valetudo_snapshot.png already exists. This would also prevent unwanted overwriting of valetudo_snapshot.png in the highly unlikely event that the user already has this file in their system unrelated to this integration.

**Comment by sca075** [comment]
@Skeletorjus yes.. I thought about this improvement too and actually I did made same try and I hope soon this will become possible. What I want to do is the following: We configure the vacuum entity for the reason that the camera name should be the same as the vacuum entity.. example: vacuum.robot_1 = camera.robot_1, vacuum.robot_2 = camera.robot_2 this will allow easily to create a new camera for each robot. As now you can do the same with the HA interface you need to input a camera name and change the entity_id of the camera once configured. About the snapshot name... yep it is a great idea.. will rename the file with the vacuum name so that it will be possible to get the snapshot of each single vacuum.. I would ask you just to be patient and wait for v1.2.0 that should come within end of next week. Thanks a lot for your active cooperation and highlights :)

**Comment by sca075** [comment]
@Skeletorjus I did modify the snapshots names.. so the camera will now save the vacuum topic (Valetudo guys have some work to do on the vacuum name for HA MQTT so this is to avoid that there will be some brake down if something will change in the future). I'm going to implement also a camera attribute that will give the file name of the image saved, therefore it will be more easy to automate the snapshots sequence in HA.

**Comment by sca075** [comment]
@Skeletorjus we are almost there.. I found the way to setup the camera entity name and id automatically from the MQTT topic. There is still to adjust the config_flow.. because on there the entity name is not updated yet.. I think with the next release the multiple cameras can be easily setup. I hope everything is going well with the v1.2.0.. peaty that I didn't wait one more day before releasing it :)

**Comment by sca075** [comment]
@Skeletorjus I released v1.3.0.beta.1, when you are ready with your generation1 (that is actually what I use) please fill free to comment, open issues and or make other improvement proposals. It will be a magior change.. I hope to fake a second vacuum with a little of effort and test the camera too. I'm going to close this per there is an official announcement  done for the beta.
